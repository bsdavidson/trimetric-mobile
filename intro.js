import React, {Component} from "react";
import {
  StyleSheet,
  Text,
  Animated,
  Image,
  View,
  TouchableOpacity,
  Switch,
  Dimensions,
  StatusBar,
  Button
} from "react-native";
import {connect} from "react-redux";

import Loading from "./loading";
import layersIcon from "./assets/layers.png";
import busIcon from "./assets/bus.png";
import mapIcon from "./assets/map.png";
import trainIcon from "./assets/tram.png";
import stopIcon from "./assets/stop.png";
import {updateLayerVisibility} from "./actions";

class Intro extends Component {
  constructor(props) {
    super(props);

    this.state = {
      visible: true,
      screenIndex: 0,
      scrollTween: new Animated.Value(0)
    };
    this.screenWidth = Dimensions.get("window").width;
    this.handleResetPress = this.handleResetPress.bind(this);
    this.handleTogglePress = this.handleTogglePress.bind(this);
    this.handleNextPress = this.handleNextPress.bind(this);
  }

  componentDidMount() {
    this.setScreenIndex(0);
  }

  componentDidUpdate() {
    if (this.state.screenIndex > 3 && this.props.loaded && this.state.visible) {
      this.setState({
        visible: false
      });
    }
  }

  handleTogglePress() {
    this.setState({visible: !this.state.visible});
  }

  handleNextPress() {
    this.setScreenIndex(this.state.screenIndex + 1);
  }

  handleResetPress() {
    this.setScreenIndex(0);
  }

  setScreenIndex(screenIndex) {
    this.state.scrollTween.setValue(0);
    Animated.timing(this.state.scrollTween, {
      toValue: 1,
      duration: 300,
      useNativeDriver: true
    }).start();

    this.setState({
      screenIndex
    });
  }

  renderTitle() {
    return (
      <View style={styles.title}>
        <View style={styles.titleTextView}>
          <Text style={styles.titleText}>trimetric</Text>
        </View>
        <Loading />
      </View>
    );
  }

  renderScreens(screenWidth) {
    return (
      <Animated.View
        style={{
          width: screenWidth * 4,
          height: 300,
          transform: [
            {
              translateX: this.state.scrollTween.interpolate({
                inputRange: [0, 1],
                outputRange: [
                  (this.state.screenIndex - 1) * screenWidth * -1,
                  this.state.screenIndex * screenWidth * -1
                ]
              })
            }
          ],
          flex: 1,
          flexDirection: "row"
        }}>
        <View style={[styles.screen, {width: screenWidth}]}>
          <View style={styles.screenPage}>
            <Image style={styles.mapImage} source={mapIcon} />
            <Text style={[styles.screenText, {marginTop: 10}]}>
              Trimetric is an open source visualization that lets you watch
              what's going on in Portland's transit system in real-time.
            </Text>
            <View style={styles.nextButton}>
              <Button title="Next" onPress={this.handleNextPress} />
            </View>
          </View>
        </View>

        <View style={[styles.screen, {width: screenWidth}]}>
          <View style={styles.screenPage}>
            <View style={{flexDirection: "row", alignItems: "center"}}>
              <Image
                resizeMode={"contain"}
                style={styles.vehicleImage}
                source={busIcon}
              />
              <Image
                resizeMode={"contain"}
                style={styles.vehicleImage}
                source={trainIcon}
              />
            </View>
            <Text style={styles.screenTextHeader}>
              These icons are vehicles.
            </Text>
            <Text style={[styles.screenText]}>
              Tapping on them will follow them as they drive around&nbsp;town.
            </Text>
            <View style={styles.nextButton}>
              <Button title="Next" onPress={this.handleNextPress} />
            </View>
          </View>
        </View>
        <View style={[styles.screen, {width: screenWidth}]}>
          <View style={styles.screenPage}>
            <Image
              resizeMode={"contain"}
              style={styles.stopImage}
              source={stopIcon}
            />
            <Text style={[styles.screenTextHeader]}>
              These icons are stops.
            </Text>
            <Text style={[styles.screenText]}>
              Tapping on them will show you upcoming arrivals and how far away
              they are from the&nbsp;stop.
            </Text>
            <View style={styles.nextButton}>
              <Button title="Next" onPress={this.handleNextPress} />
            </View>
          </View>
        </View>
        <View style={[styles.screen, {width: screenWidth}]}>
          <View style={styles.screenPage}>
            <Image
              resizeMode={"contain"}
              style={styles.layersImage}
              source={layersIcon}
            />
            <Text style={[styles.screenTextHeader, {marginTop: 0}]}>
              This is the layers menu.
            </Text>
            <Text style={[styles.screenText]}>
              The map will show you many details, which you can show or hide
              using this&nbsp;menu.
            </Text>
            <View style={styles.nextButton}>
              <Button title="Okay, let's go!" onPress={this.handleNextPress} />
            </View>
          </View>
        </View>
      </Animated.View>
    );
  }
  render() {
    const screenWidth = this.props.dimensions.window.width;
    if (!this.state.visible) {
      return null;
    }

    return (
      <View
        style={{
          position: "absolute",
          top: 0,
          bottom: 0,
          left: 0,
          right: 0,
          height: "100%",
          backgroundColor: "#34495e",
          zIndex: 99
        }}>
        <StatusBar barStyle="light-content" />
        {this.renderTitle()}
        {this.renderScreens(screenWidth)}

        <TouchableOpacity
          style={{position: "absolute", left: 10, bottom: 10}}
          onPress={this.handleResetPress}>
          <Text>Reset</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

function mapStateToProps(state) {
  return {
    layerVisibility: state.layerVisibility,
    dimensions: state.dimensions,
    loaded: state.loaded
  };
}

function mapDispatchToProps(dispatch) {
  return {
    onUpdateLayerVisibility: (layerName, value) => {
      dispatch(updateLayerVisibility(layerName, value));
    }
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Intro);

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#FFFFFF",
    borderRadius: 6,
    elevation: 4,
    position: "absolute",
    padding: 10,
    right: 35,
    left: 35,
    maxWidth: 600,
    shadowColor: "#000",
    shadowOffset: {width: 0, height: -8},
    shadowOpacity: 0.4,
    shadowRadius: 6,
    top: 105,
    zIndex: 100
  },
  screen: {
    alignItems: "center",
    flex: 1,
    padding: 30,
    paddingBottom: 50,
    flexDirection: "column",
    justifyContent: "flex-end"
  },
  screenText: {
    fontSize: 16,
    marginTop: 15,
    color: "#222222",
    textAlign: "center"
  },
  screenTextHeader: {
    fontWeight: "bold",
    marginTop: 15,
    fontSize: 16,
    color: "#222222",
    textAlign: "center"
  },
  screenPage: {
    borderRadius: 12,
    backgroundColor: "#efefef",
    flex: 0,
    padding: 15,
    paddingLeft: 15,
    paddingRight: 15,
    alignItems: "center",
    shadowColor: "#000",
    shadowOffset: {width: 0, height: 8},
    shadowOpacity: 0.4,
    shadowRadius: 6
  },
  title: {
    flex: 1,
    alignItems: "center",
    justifyContent: "flex-end"
  },
  titleText: {
    flex: 0,
    color: "#ecf0f1",
    fontFamily: "Allerta Stencil",
    fontSize: 50,
    marginBottom: 20
  },
  titleTextView: {
    flex: 1,
    justifyContent: "flex-end"
  },
  nextButton: {
    marginTop: 5
  },
  vehicleImage: {
    height: 60,
    width: 60,
    margin: 5,
    marginTop: 2,
    flex: 1
  },
  mapImage: {
    height: 75,
    width: 200,
    margin: 5,
    padding: 10,

    marginTop: 2,
    borderRadius: 6,
    borderWidth: 1,
    borderColor: "#AAAAAA"
  },
  stopImage: {
    height: 60,
    width: 60,
    margin: 5,
    marginTop: 2
  },
  layersImage: {
    height: 80,
    width: 80,
    margin: 5,
    marginTop: 2
  }
});
