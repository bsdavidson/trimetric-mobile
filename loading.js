import React, {Component} from "react";
import {View, Text, ActivityIndicator, StyleSheet} from "react-native";
import {getVehiclePoints, getStopPoints} from "./selectors";
import {connect} from "react-redux";

import {setLoadingStatusLoaded} from "./actions";

export class Loading extends Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: true
    };

    this.loadingStatus = 0;
  }

  componentDidUpdate() {
    if (!this.state.loading) {
      return;
    }
    if (!this.props.totals) {
      return;
    }
    let thingsLoaded =
      this.props.stopPoints.features.length +
      (this.props.routeShapes && this.props.routeShapes.length);
    let totalThingsToLoad =
      this.props.totals.stops + this.props.totals.route_shapes;

    this.loadingStatus = Math.min(
      Math.floor(thingsLoaded / totalThingsToLoad * 100),
      100
    );

    if (this.props.stopPoints.features.length < this.props.totals.stops) {
      return;
    }

    if (
      !this.props.routeShapes ||
      this.props.routeShapes.length < this.props.totals.route_shapes
    ) {
      return;
    }

    if (!this.props.receivedVehicles) {
      return;
    }

    this.props.onLoadComplete();
    this.setState({
      loading: false
    });
  }

  render() {
    if (!this.props.connected) {
      return (
        <View style={styles.container}>
          <ActivityIndicator
            animating={this.state.loading}
            size="small"
            color="#FFFFFF"
          />
          <Text style={[styles.text]}>Connecting to server...</Text>
        </View>
      );
    }

    return (
      <View style={styles.container}>
        <ActivityIndicator
          animating={this.state.loading}
          size="small"
          color="#FFFFFF"
        />

        <Text style={[styles.text, {opacity: this.state.loading ? 1 : 0}]}>
          Loading GTFS data...
          {/* <Text style={{color: "#FFFFFF"}}>{this.loadingStatus}%</Text> */}
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    alignItems: "flex-start",
    flex: 1,
    flexDirection: "row",
    justifyContent: "center"
  },
  text: {
    color: "#ecf0f1",
    marginLeft: 10
  }
});

function mapDispatchToProps(dispatch) {
  return {
    onLoadComplete: items => {
      dispatch(setLoadingStatusLoaded());
    }
  };
}

function mapStateToProps(state) {
  return {
    connected: state.connected,
    receivedVehicles: state.receivedVehicles,
    routeShapes: state.routeShapes,
    stopPoints: getStopPoints(state),
    totals: state.totals
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Loading);
