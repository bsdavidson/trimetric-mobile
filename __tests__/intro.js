import "react-native";
import React from "react";
import {Intro} from "../intro";
import {Loading} from "../loading";
import renderer from "react-test-renderer";

jest.mock("../loading", () => {
  return "Loading";
});

jest.mock("Animated");

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Intro
        dimensions={{window: {width: 400, height: 600}}}
        layerVisibility={{}}
        loaded={false}
        onSeenIntro={() => {}}
        onUpdateLayerVisibility={() => {}}
        seenIntro={false}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
