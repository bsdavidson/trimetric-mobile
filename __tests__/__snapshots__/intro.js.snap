// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#34495e",
      "bottom": 0,
      "height": "100%",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 99,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ecf0f1",
            "flex": 0,
            "fontFamily": "Allerta Stencil",
            "fontSize": 50,
            "marginBottom": 20,
          }
        }
      >
        trimetric
      </Text>
    </View>
    <Loading />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "height": 300,
        "transform": Array [
          Object {
            "translateX": 1,
          },
        ],
        "width": 1600,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-end",
            "padding": 30,
            "paddingBottom": 50,
          },
          Object {
            "width": 400,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#efefef",
            "borderRadius": 12,
            "flex": 0,
            "padding": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 8,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 6,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../assets/map.png",
            }
          }
          style={
            Object {
              "borderColor": "#AAAAAA",
              "borderRadius": 6,
              "borderWidth": 1,
              "height": 75,
              "margin": 5,
              "marginTop": 2,
              "padding": 10,
              "width": 200,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#222222",
                "fontSize": 16,
                "marginTop": 15,
                "textAlign": "center",
              },
              Object {
                "marginTop": 10,
              },
            ]
          }
        >
          Trimetric is an open source visualization that lets you watch what's going on in Portland's transit system in real-time.
        </Text>
        <View
          style={
            Object {
              "marginTop": 5,
            }
          }
        >
          <View
            accessibilityComponentType="button"
            accessibilityLabel={undefined}
            accessibilityTraits={
              Array [
                "button",
              ]
            }
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                undefined,
                Object {
                  "opacity": Value {
                    "value": 1,
                  },
                },
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={undefined}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "padding": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Next
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-end",
            "padding": 30,
            "paddingBottom": 50,
          },
          Object {
            "width": 400,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#efefef",
            "borderRadius": 12,
            "flex": 0,
            "padding": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 8,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 6,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Image
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../assets/bus.png",
              }
            }
            style={
              Object {
                "flex": 1,
                "height": 60,
                "margin": 5,
                "marginTop": 2,
                "width": 60,
              }
            }
          />
          <Image
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../assets/tram.png",
              }
            }
            style={
              Object {
                "flex": 1,
                "height": 60,
                "margin": 5,
                "marginTop": 2,
                "width": 60,
              }
            }
          />
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#222222",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginTop": 15,
              "textAlign": "center",
            }
          }
        >
          These icons are vehicles.
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#222222",
                "fontSize": 16,
                "marginTop": 15,
                "textAlign": "center",
              },
            ]
          }
        >
          Tapping on them will follow them as they drive aroundÂ town.
        </Text>
        <View
          style={
            Object {
              "marginTop": 5,
            }
          }
        >
          <View
            accessibilityComponentType="button"
            accessibilityLabel={undefined}
            accessibilityTraits={
              Array [
                "button",
              ]
            }
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                undefined,
                Object {
                  "opacity": Value {
                    "value": 1,
                  },
                },
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={undefined}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "padding": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Next
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-end",
            "padding": 30,
            "paddingBottom": 50,
          },
          Object {
            "width": 400,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#efefef",
            "borderRadius": 12,
            "flex": 0,
            "padding": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 8,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 6,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../assets/stop.png",
            }
          }
          style={
            Object {
              "height": 60,
              "margin": 5,
              "marginTop": 2,
              "width": 60,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#222222",
                "fontSize": 16,
                "fontWeight": "bold",
                "marginTop": 15,
                "textAlign": "center",
              },
            ]
          }
        >
          These icons are stops.
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#222222",
                "fontSize": 16,
                "marginTop": 15,
                "textAlign": "center",
              },
            ]
          }
        >
          Tapping on them will show you upcoming arrivals and how far away they are from theÂ stop.
        </Text>
        <View
          style={
            Object {
              "marginTop": 5,
            }
          }
        >
          <View
            accessibilityComponentType="button"
            accessibilityLabel={undefined}
            accessibilityTraits={
              Array [
                "button",
              ]
            }
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                undefined,
                Object {
                  "opacity": Value {
                    "value": 1,
                  },
                },
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={undefined}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "padding": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Next
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-end",
            "padding": 30,
            "paddingBottom": 50,
          },
          Object {
            "width": 400,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#efefef",
            "borderRadius": 12,
            "flex": 0,
            "padding": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 8,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 6,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../assets/layers.png",
            }
          }
          style={
            Object {
              "height": 80,
              "margin": 5,
              "marginTop": 2,
              "width": 80,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#222222",
                "fontSize": 16,
                "fontWeight": "bold",
                "marginTop": 15,
                "textAlign": "center",
              },
              Object {
                "marginTop": 0,
              },
            ]
          }
        >
          This is the layers menu.
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#222222",
                "fontSize": 16,
                "marginTop": 15,
                "textAlign": "center",
              },
            ]
          }
        >
          The map will show you many details, which you can show or hide using thisÂ menu.
        </Text>
        <View
          style={
            Object {
              "marginTop": 5,
            }
          }
        >
          <View
            accessibilityComponentType="button"
            accessibilityLabel={undefined}
            accessibilityTraits={
              Array [
                "button",
              ]
            }
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                undefined,
                Object {
                  "opacity": Value {
                    "value": 1,
                  },
                },
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={undefined}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "padding": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Okay, let's go!
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
